src/main/java/kitEditor/KitEditor.java

```diff
@@ -1132,7 +1140,62 @@ private void trimAllSamples() {
                 "Done",
                 JOptionPane.INFORMATION_MESSAGE);
     }
+
+    private void duplicateSample(Sample sample) {
+        if (sample == null) {
+          return;
+        }
+        int dest = firstFreeSampleSlot();
+        Sample firstSample = sample;
+        Sample dupeSample;
+        if (dest != -1) {
+            // copy sample data
+            try {
+            dupeSample = Sample.dupeSample(firstSample);
+            dupeSample.reload(halfSpeed.isSelected());
+            } catch (Exception e) {
+                showFileErrorMessage(e);
+                return;
+            }
+            samples[selectedBank][dest] = dupeSample;
```

Comment:
Codestyle note : this is the kind of changes I'd place at the end of the whole `dupleSample` rename/trim process, just to avoid having to reset `samples[selectedBank][dest]` to `null` if part of the process went wrong. It makes the code more manageable in the future as there'll be less chances that a refactor will leave a half-broken sample in the `samples` array (got fun crashes like this in the past with multi-threaded code).

For future reference, this is the idea of commits in database software, either do all  the changes at once or not, there is no in-betweens.